pkg/authorization/authorizer/scope/converter.go

```diff
@@ -48,6 +49,40 @@ func ScopesToRules(scopes []string, namespace string, clusterPolicyGetter client
 	return rules, kutilerrors.NewAggregate(errors)
 }
 
+// ScopesToVisibleNamespaces returns a list of namespaces that the provided scopes have "get" access to.
+// This exists only to support efficiently list/watch of projects (ACLed namespaces)
+func ScopesToVisibleNamespaces(scopes []string, clusterPolicyGetter client.ClusterPolicyLister) (sets.String, error) {
+	if len(scopes) == 0 {
+		return sets.NewString("*"), nil
+	}
+
+	visibleNamespaces := sets.String{}
+
+	errors := []error{}
+	for _, scope := range scopes {
+		found := false
+
+		for _, evaluator := range ScopeEvaluators {
+			if evaluator.Handles(scope) {
+				found = true
+				allowedNamespaces, err := evaluator.ResolveGettableNamespaces(scope, clusterPolicyGetter)
+				if err != nil {
+					errors = append(errors, err)
+					continue
+				}
+
+				visibleNamespaces.Insert(allowedNamespaces...)
```

Comment:
> if we now contain *, return early?

That assumes this knows how the results will be used.  I'd rather not, but if you feel strongly I won't fight.